Jint/Collections/RefStack.cs

```diff
@@ -0,0 +1,198 @@
+﻿#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Jint.Runtime;
+
+namespace Jint.Collections
+{
+    /// <summary>
+    /// Stack for struct types.
+    /// </summary>
+    internal sealed class RefStack<T> : IEnumerable<T> where T : struct
```

Comment:
Tell me more about that!

Reply:
Basically just optimized to return refs without copies (like structs usually get copied with generic .NET data structures). Faster and simpler for specific use case.